### ISSUE TITLE
[Site Assets] Select All checkbox is not working

### ISSUE THREAD

**Issue by mikebigun** [dnnsoftware/Dnn.Platform#3250]
## Description of bug
Select All checkbox on toolbar is not working properly. When user clicks on `Select All` and then navigates to different folder, checkbox remains checked, but no one item is selected. 

## Steps to reproduce
List the steps to reproduce the behavior:
1. Login as admin
2. Go to Persona Bar -> Site Assets
3. Navigate to `Root` folder
4. Click on `Select All`
5. Navigate to `Images` folder

## Current result
When user navigates to different folder, checkbox remains checked. You can observer that clicking on this checkbox it does not reflect its state correctly. 

## Expected result
`Select All` checkbox should work properly. It should reflect its state correctly. When `Select All` is checked, all items should be selected, otherwise deselected. 

## Screenshots

![image](https://user-images.githubusercontent.com/22524011/68304046-a96c2300-00ad-11ea-8a1b-7d2444bcaac1.png)

![image](https://user-images.githubusercontent.com/22524011/68304096-c3a60100-00ad-11ea-8f5d-6f8e66ef5df3.png)

## Affected version

* [x] 10.0.0 alpha build
* [x] 9.4.2 alpha build
* [x] 9.4.1 latest supported release

## Affected browser

* [x] Chrome
* [x] Firefox
* [x] Safari
* [x] Internet Explorer
* [x] Edge

**Comment by valadas** [comment]
I am not able to reproduce this in 9.4.2 RC1, is there anything else special in order to reproduce this issue? Does it happen with the default theme? Do you have any browser console errors?

**Comment by valadas** [comment]
Just tried it on 9.4.0 and 9.4.1 too and not able to reproduce

**Comment by thabaum** [comment]
I could not reproduce either
![image](https://user-images.githubusercontent.com/13577556/68537359-159d8f80-0317-11ea-810b-3479efd1842c.png)
![image](https://user-images.githubusercontent.com/13577556/68537365-24844200-0317-11ea-924d-186b1ba6bee5.png)

I do notice your view is different then stock.  You have large thumbnails when I have detailed view going.

**Comment by thabaum** [comment]
reproduced
![image](https://user-images.githubusercontent.com/13577556/68537384-7331dc00-0317-11ea-8c38-56d210390081.png)

**Comment by thabaum** [comment]
reproduced.  

Icon View is what sets off the bug.  I just had to click on images, select all images and click on the images folder again it would refresh looking like this:

![image](https://user-images.githubusercontent.com/13577556/68539371-79d04b80-0337-11ea-8413-538ff2916888.png)

If you use detailed view it will de-select the checkbox as expected.  I verified after you do check the select all checkbox while using Icon View in Site Assets navigate into any folder with items or not... and then simply change to another folder the checkbox select all remains selected.  

A work around is to use the other view for now but this needs to be looked into along with your solution further IMO.
